engine/src/cmd/damageable.cpp

```diff
@@ -495,28 +494,56 @@ void Damageable::Destroy() {
 }
 
 
+// We only support 2 and 4 facet shields ATM
+// This doesn't have proper checks
+float ShieldData(const Damageable *unit, int facet_index) {
+    return (unit->shield->facets[facet_index].health)/
+            (unit->shield->facets[facet_index].max_health);
+}
+
+
 float Damageable::FShieldData() const
 {
-    Damageable *damageable = const_cast<Damageable*>(this);
-    return static_cast<DamageableLayer>(damageable->GetShieldLayer()).GetPercent(FacetName::front);
+    switch(shield->number_of_facets) {
+    case 2:
```

Comment:
would it be possible to use enums instead of 2/4 in these?

Reply:
I'd leave that to the next PR. It's better than it has been and let's leave it at that.